src/okta/services/auth-state.service.ts

```diff
@@ -28,6 +31,42 @@ export class OktaAuthStateService implements OnDestroy {
     this.oktaAuth.authStateManager.unsubscribe(this.updateAuthState);
   }
 
+  // Observes as true when any group input can match groups from user claims 
+  hasAnyGroups(groups: Groups): Observable<boolean> {
+    return this.authState$.pipe(
+      mergeMap(async ({ isAuthenticated, idToken }) => {
+        // return false when not authenticated or openid is not in scopes
+        if (!isAuthenticated || !idToken) {
+          return false;
+        }
+
+        // transform inputs to consistent object format
+        if (typeof groups === 'string') {
+          groups = { groups: [groups] };
+        }
+        if (Array.isArray(groups)) {
+          groups = { groups };
+        }
+
+        const key = Object.keys(groups)[0];
+        const value = groups[key];
+
+        // groups or custom claims is avaiable in idToken
+        if (idToken.claims[key]) {
```

Comment:
Would verifying ID token before checking the claims be useful for this feature?

Reply:
It's checked at the beginning of this method. around line 39